cpp/include/rapidsmpf/buffer/buffer.hpp

```diff
@@ -122,12 +124,27 @@ class Buffer {
         );
     }
 
-    /// @brief Buffer has a move ctor but no copy or assign operator.
-    Buffer(Buffer&&) = default;
+    /**
+     * @brief Check if the device memory operation has completed.
+     *
+     * @return true if the device memory operation has completed or no device
+     * memory operation was performed, false if it is still in progress.
+     */
+    [[nodiscard]] bool is_ready() const;
+
+    /// @brief Delete move and copy constructors and assignment operators.
+    Buffer(Buffer&&) = delete;
```

Comment:
I presume the move constructor was intended for https://github.com/rapidsai/rapidsmpf/pull/225/files#diff-43d52141d27f2bb2eef5135fcdefc2e48844716ffc6dd4939cec38875a00db04L91-R114, however, it did not move and instead constructs/destroys/reconstruct `Buffer`, which invalidated the `cuda_event_`. Therefore, those changes were made to ensure we only construct `Buffer` once, and thus the move constructor is unnecessary here as well.